src/lib/pages/proposal/whitelist/index.tsx

```diff
@@ -0,0 +1,428 @@
+
+import {
+  useFabricateFee,
+  useSimulateFeeQuery,
+  useSubmitProposalTx,
+} from "lib/app-provider";
+import { AddressInput } from "lib/components/AddressInput";
+import { AssignMe } from "lib/components/AssignMe";
+import { ConnectWalletAlert } from "lib/components/ConnectWalletAlert";
+import { ErrorMessageRender } from "lib/components/ErrorMessageRender";
+import { EstimatedFeeRender } from "lib/components/EstimatedFeeRender";
+import { ControllerInput, ControllerTextarea } from "lib/components/forms";
+import { CustomIcon } from "lib/components/icon";
+import PageContainer from "lib/components/PageContainer";
+import { StickySidebar } from "lib/components/StickySidebar";
+import {
+  getMaxProposalTitleLengthError,
+  MAX_PROPOSAL_TITLE_LENGTH,
+} from "lib/data/proposalWhitelist";
+import { useTxBroadcast } from "lib/providers/tx-broadcast";
+import { AmpEvent, AmpTrack } from "lib/services/amplitude";
+import { useGovParams } from "lib/services/proposalService";
+import type { Addr } from "lib/types";
+import { composeSubmitWhitelistProposalMsg, d2Formatter } from "lib/utils";
+
+import { AssetBox, Footer } from "./components";
+import { getAlert } from "./utils";
+
+interface WhiteListState {
+  title: string;
+  description: string;
+  addresses: { address: Addr }[];
+  initialDeposit: Coin;
+}
+
+const defaultValues: WhiteListState = {
+  title: "",
+  description: "",
+  addresses: [{ address: "" as Addr }],
+  initialDeposit: { denom: "", amount: "" } as Coin,
+};
+
+const Whitelisting = observer(() => {
+  const router = useRouter();
+  const { address: walletAddress = "" } = useWallet();
+  const fabricateFee = useFabricateFee();
+  const { data: govParams } = useGovParams();
+  const submitProposalTx = useSubmitProposalTx();
+  const { broadcast } = useTxBroadcast();
+  const {
+    control,
+    watch,
+    setValue,
+    reset,
+    formState: { errors: formErrors },
+  } = useForm<WhiteListState>({
+    defaultValues,
+    mode: "all",
+  });
+
+  const [estimatedFee, setEstimatedFee] = useState<StdFee>();
+  const [simulateError, setSimulateError] = useState<string>();
+  const [processing, setProcessing] = useState(false);
+  const { title, description, addresses, initialDeposit } = watch();
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: "addresses",
+  });
+
+  const addressesArray = addresses.map((addressObj) => addressObj.address);
+  const enabledTx = useMemo(
+    () =>
+      Boolean(
+        walletAddress &&
+          title &&
+          description &&
+          addressesArray.length > 0 &&
+          initialDeposit.amount &&
+          !Object.keys(formErrors).length
+      ),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [
+      walletAddress,
+      title,
+      description,
+      addressesArray,
+      initialDeposit.amount,
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      Object.keys(formErrors),
```

Comment:
can you just use `formErrors` here? or can we create a variable for formErrors keys and use it here instead? so we can remove both `eslint`s